src/main/java/com/huy/mse/brand/BrandServiceImpl.java

```diff
@@ -27,22 +27,23 @@ public List<BrandDto> getAllBrands() {
     public BrandDto getBrandById(long id) {
         return brandRepository.findById(id)
                 .map(brandMapper::toDto)
-                .orElse(null);
+                .orElseThrow(() -> new RuntimeException("Brand not found"));
```

Comment:
Suy nghĩ nếu chỉ throw exception thì những th như k tìm thấy, kết nối lỗi, thì xử lý thế nào

Reply:
Không tìm thấy thì vẫn trả về http 400 với message là brand not found, thì bên client vẫn xử lí là không tìm cho cả 2 trường hợp. Trường hợp kết nỗi lỗi thì sẽ bay ra lỗi 500 internal server error á anh.